synapse/storage/databases/main/metrics.py

```diff
@@ -12,17 +12,35 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import typing
-from collections import Counter
 
-from synapse.metrics import BucketCollector
+from synapse.metrics import GaugeBucketCollector
 from synapse.metrics.background_process_metrics import run_as_background_process
 from synapse.storage._base import SQLBaseStore
 from synapse.storage.database import DatabasePool
 from synapse.storage.databases.main.event_push_actions import (
     EventPushActionsWorkerStore,
 )
 
+# Collect metrics on the number of forward extremities that exist.
+_extremities_collecter = GaugeBucketCollector(
+    "synapse_forward_extremities",
+    "Number of rooms on the server with the given number of forward extremities"
+    " or fewer",
+    buckets=[1, 2, 3, 5, 7, 10, 15, 20, 50, 100, 200, 500],
```

Comment:
why would we ever expect there to be significantly more than 10 forward extremities (or whatever the dummy threshold is)?

Reply:
well, this is what we have today...

There are a few reasons we might end up with more:
 * there are no users in the room who can send an event, for reasons of power-levels or consent
 * we can't keep up with the amount of divergence in the dag (#7792 was a big cause of this, but is hopefully now fixed)